### ISSUE TITLE
s3queue fails to continue ingesting properly once caught up.

### ISSUE THREAD

**Issue by deltarod** [ClickHouse/ClickHouse#64710]
**Version**

```
SELECT version();
24.4.1.2088
```

**Describe what's wrong**

What I am currently trying to do is instead of going straight to clickhouse, put the records into s3 first and then use s3queue to stream that data into clickhouse. Upon initial creation of the s3queue table, it loads perfectly. Once the s3queue "catches up" it then stops loading correctly. Dropping and recreating the s3queue table seems to make it catch up. It seems like it is still loading but is failing to load all the data.

My sql defs are essentially this. I can provide more details if needed but im not doing anything weird so it should function the same?: https://clickhouse.com/docs/en/engines/table-engines/integrations/s3queue#description

The s3 path looks like `s3://bucket-name/folder/tablename/*/part-*`

**How to reproduce**

1. Have an S3 Producer which creates JSONEachRow format files.
2. create table with engine def that looks like this:
`ENGINE = S3('s3://bucket/folder/tablename/*/part-*', 'JSONEachRow');`
3. wait until load catches up
4. s3queue either stops loading or 

**Expected behavior**

> It continues to load

**Error message and/or stacktrace**

I'm not sure if I can directly link these error messages to this problem but here are some potential relevant lines from clickhouse-server.err.log

Upon creation/recreation of the s3queue table, I see this error

```2024.05.31 20:17:34.583624 [ 83330 ] {6eb15e56-cd70-4ce7-8727-3b68c5cb9227} <Warning> AwsAuthSTSAssumeRoleWebIdentityCredentialsProvider: Token file must be specified to use STS AssumeRole web identity creds provider.```

These errors come after the table catches up, not sure if they are relevant or caused by something else occuring.
```
2024.05.31 20:20:21.208851 [ 82616 ] {} <Warning> CgroupsMemoryUsageObserver: Exceeded soft memory limit (99.74 GiB)
2024.05.31 20:20:27.467097 [ 82726 ] {} <Error> MySQLDictionarySource: Cannot find table in SHOW TABLE STATUS result.
2024.05.31 20:20:27.472510 [ 82864 ] {} <Error> MySQLDictionarySource: Cannot find table in SHOW TABLE STATUS result.
2024.05.31 20:21:21.515857 [ 82616 ] {} <Warning> CgroupsMemoryUsageObserver: Exceeded soft memory limit (99.74 GiB)
2024.05.31 20:23:36.798665 [ 82616 ] {} <Warning> CgroupsMemoryUsageObserver: Exceeded hard memory limit (105.28 GiB)
2024.05.31 20:24:36.799838 [ 82616 ] {} <Warning> CgroupsMemoryUsageObserver: Exceeded hard memory limit (105.28 GiB)
2024.05.31 20:25:21.800730 [ 82616 ] {} <Warning> CgroupsMemoryUsageObserver: Exceeded hard memory limit (105.28 GiB)
2024.05.31 20:26:21.801755 [ 82616 ] {} <Warning> CgroupsMemoryUsageObserver: Exceeded hard memory limit (105.28 GiB)
2024.05.31 20:30:52.696825 [ 82726 ] {} <Error> MySQLDictionarySource: Cannot find table in SHOW TABLE STATUS result.
2024.05.31 20:30:52.701387 [ 82797 ] {} <Error> MySQLDictionarySource: Cannot find table in SHOW TABLE STATUS result.
2024.05.31 20:31:36.807888 [ 82616 ] {} <Warning> CgroupsMemoryUsageObserver: Exceeded hard memory limit (105.28 GiB)
2024.05.31 20:31:47.706719 [ 82726 ] {} <Error> MySQLDictionarySource: Cannot find table in SHOW TABLE STATUS result.
2024.05.31 20:31:47.711682 [ 83298 ] {} <Error> MySQLDictionarySource: Cannot find table in SHOW TABLE STATUS result.
2024.05.31 20:32:06.808499 [ 82616 ] {} <Warning> CgroupsMemoryUsageObserver: Exceeded hard memory limit (105.28 GiB)
```

**Additional context**

Link to relevant chat in slack
https://clickhousedb.slack.com/archives/CU478UEQZ/p1717182119242689

**Comment by deltarod** [comment]
I posted this in slack but putting here for extra context for anyone else:
```
2024.06.03 18:23:17.868481 [ 702 ] {} <Information> AWSClient: Failed to make request to: https://removed: Poco::Exception. Code: 1000, e.code() = 0, No message received, Stack trace (when copying this message, always include the lines below):

0. Poco::Net::HTTPResponse::read(std::basic_istream<char, std::char_traits<char>>&) @ 0x0000000014c8570b
1. Poco::Net::HTTPClientSession::receiveResponse(Poco::Net::HTTPResponse&) @ 0x0000000014c7bf08
2. DB::EndpointConnectionPool<Poco::Net::HTTPSClientSession>::PooledConnection::receiveResponse(Poco::Net::HTTPResponse&) @ 0x000000000cdc5249
3. DB::S3::PocoHTTPClient::makeRequestInternalImpl(Aws::Http::HttpRequest&, DB::ProxyConfiguration const&, std::shared_ptr<DB::S3::PocoHTTPResponse>&, Aws::Utils::RateLimits::RateLimiterInterface*, Aws::Utils::RateLimits::RateLimiterInterface*) const @ 0x000000000f997f64
4. DB::S3::PocoHTTPClient::MakeRequest(std::shared_ptr<Aws::Http::HttpRequest> const&, Aws::Utils::RateLimits::RateLimiterInterface*, Aws::Utils::RateLimits::RateLimiterInterface*) const @ 0x000000000f994132
5. std::shared_ptr<Aws::Http::HttpResponse> smithy::components::tracing::TracingUtils::MakeCallWithTiming<std::shared_ptr<Aws::Http::HttpResponse>>(std::function<std::shared_ptr<Aws::Http::HttpResponse> ()>, String const&, smithy::components::tracing::Meter const&, std::map<String, String, std::less<String>, std::allocator<std::pair<String const, String>>>&&, String const&) @ 0x0000000014dff26f
6. Aws::Client::AWSClient::AttemptExhaustively(Aws::Http::URI const&, Aws::AmazonWebServiceRequest const&, Aws::Http::HttpMethod, char const*, char const*, char const*) const @ 0x0000000014df1d37
7. Aws::Client::AWSXMLClient::MakeRequest(Aws::Http::URI const&, Aws::AmazonWebServiceRequest const&, Aws::Http::HttpMethod, char const*, char const*, char const*) const @ 0x0000000014e0f4f6
8. Aws::Utils::Outcome<Aws::S3::Model::ListObjectsV2Result, Aws::S3::S3Error> std::__function::__policy_invoker<Aws::Utils::Outcome<Aws::S3::Model::ListObjectsV2Result, Aws::S3::S3Error> ()>::__call_impl<std::__function::__default_alloc_func<Aws::S3::S3Client::ListObjectsV2(Aws::S3::Model::ListObjectsV2Request const&) const::$_0, Aws::Utils::Outcome<Aws::S3::Model::ListObjectsV2Result, Aws::S3::S3Error> ()>>(std::__function::__policy_storage const*) @ 0x0000000014fa2242
9. Aws::S3::S3Client::ListObjectsV2(Aws::S3::Model::ListObjectsV2Request const&) const @ 0x0000000014f11648
10. DB::S3::Client::ListObjectsV2(DB::S3::ExtendedRequest<Aws::S3::Model::ListObjectsV2Request>&) const @ 0x000000000f95d257
11. Aws::Utils::Outcome<Aws::S3::Model::ListObjectsV2Result, Aws::S3::S3Error> std::__function::__policy_invoker<Aws::Utils::Outcome<Aws::S3::Model::ListObjectsV2Result, Aws::S3::S3Error> ()>::__call_impl<std::__function::__default_alloc_func<DB::StorageS3Source::DisclosedGlobIterator::Impl::listObjectsAsync()::'lambda'(), Aws::Utils::Outcome<Aws::S3::Model::ListObjectsV2Result, Aws::S3::S3Error> ()>>(std::__function::__policy_storage const*) @ 0x00000000119af609
12. std::__packaged_task_func<std::function<std::future<Aws::Utils::Outcome<Aws::S3::Model::ListObjectsV2Result, Aws::S3::S3Error>> (std::function<Aws::Utils::Outcome<Aws::S3::Model::ListObjectsV2Result, Aws::S3::S3Error> ()>&&, Priority)> DB::threadPoolCallbackRunner<Aws::Utils::Outcome<Aws::S3::Model::ListObjectsV2Result, Aws::S3::S3Error>, std::function<Aws::Utils::Outcome<Aws::S3::Model::ListObjectsV2Result, Aws::S3::S3Error> ()>>(ThreadPoolImpl<ThreadFromGlobalPoolImpl<false>>&, String const&)::'lambda'(std::function<Aws::Utils::Outcome<Aws::S3::Model::ListObjectsV2Result, Aws::S3::S3Error> ()>&&, Priority)::operator()(std::function<Aws::Utils::Outcome<Aws::S3::Model::ListObjectsV2Result, Aws::S3::S3Error> ()>&&, Priority)::'lambda'(), std::allocator<std::function<std::future<Aws::Utils::Outcome<Aws::S3::Model::ListObjectsV2Result, Aws::S3::S3Error>> (std::function<Aws::Utils::Outcome<Aws::S3::Model::ListObjectsV2Result, Aws::S3::S3Error> ()>&&, Priority)> DB::threadPoolCallbackRunner<Aws::Utils::Outcome<Aws::S3::Model::ListObjectsV2Result, Aws::S3::S3Error>, std::function<Aws::Utils::Outcome<Aws::S3::Model::ListObjectsV2Result, Aws::S3::S3Error> ()>>(ThreadPoolImpl<ThreadFromGlobalPoolImpl<false>>&, String const&)::'lambda'(std::function<Aws::Utils::Outcome<Aws::S3::Model::ListObjectsV2Result, Aws::S3::S3Error> ()>&&, Priority)::operator()(std::function<Aws::Utils::Outcome<Aws::S3::Model::ListObjectsV2Result, Aws::S3::S3Error> ()>&&, Priority)::'lambda'()>, Aws::Utils::Outcome<Aws::S3::Model::ListObjectsV2Result, Aws::S3::S3Error> ()>::operator()() @ 0x00000000119b7367
13. void std::__function::__policy_invoker<void ()>::__call_impl<std::__function::__default_alloc_func<std::function<std::future<Aws::Utils::Outcome<Aws::S3::Model::ListObjectsV2Result, Aws::S3::S3Error>> (std::function<Aws::Utils::Outcome<Aws::S3::Model::ListObjectsV2Result, Aws::S3::S3Error> ()>&&, Priority)> DB::threadPoolCallbackRunner<Aws::Utils::Outcome<Aws::S3::Model::ListObjectsV2Result, Aws::S3::S3Error>, std::function<Aws::Utils::Outcome<Aws::S3::Model::ListObjectsV2Result, Aws::S3::S3Error> ()>>(ThreadPoolImpl<ThreadFromGlobalPoolImpl<false>>&, String const&)::'lambda'(std::function<Aws::Utils::Outcome<Aws::S3::Model::ListObjectsV2Result, Aws::S3::S3Error> ()>&&, Priority)::operator()(std::function<Aws::Utils::Outcome<Aws::S3::Model::ListObjectsV2Result, Aws::S3::S3Error> ()>&&, Priority)::'lambda0'(), void ()>>(std::__function::__policy_storage const*) @ 0x00000000119b78ac
14. ThreadPoolImpl<ThreadFromGlobalPoolImpl<false>>::worker(std::__list_iterator<ThreadFromGlobalPoolImpl<false>, void*>) @ 0x000000000cc7b379
15. void std::__function::__policy_invoker<void ()>::__call_impl<std::__function::__default_alloc_func<ThreadFromGlobalPoolImpl<false>::ThreadFromGlobalPoolImpl<void ThreadPoolImpl<ThreadFromGlobalPoolImpl<false>>::scheduleImpl<void>(std::function<void ()>, Priority, std::optional<unsigned long>, bool)::'lambda0'()>(void&&)::'lambda'(), void ()>>(std::__function::__policy_storage const*) @ 0x000000000cc7ebda
16. void* std::__thread_proxy[abi:v15000]<std::tuple<std::unique_ptr<std::__thread_struct, std::default_delete<std::__thread_struct>>, void ThreadPoolImpl<std::thread>::scheduleImpl<void>(std::function<void ()>, Priority, std::optional<unsigned long>, bool)::'lambda0'()>>(void*) @ 0x000000000cc7d9ed
17. ? @ 0x00007f7955922609
18. ? @ 0x00007f7955847353
 (version 24.3.2.23 (official build))
2024.06.03 18:23:17.868546 [ 702 ] {} <Information> AWSClient: AWSXmlClient: HTTP response code: -1
Resolved remote host IP address: 16.182.97.209:443
Request ID: 
Exception name: 
Error message: Poco::Exception. Code: 1000, e.code() = 0, No message received (version 24.3.2.23 (official build))
0 response headers:
2024.06.03 18:23:17.868560 [ 702 ] {} <Information> AWSClient: If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.
2024.06.03 18:23:17.868566 [ 702 ] {} <Information> AWSClient: Request failed, now waiting 0 ms before attempting again.
````
the flink loader I am using does create temp files. Maybe it doesn't properly handle file renames? That's the general form of my files so ordered mode probably wouldn't work unless I'm understanding the functionality incorrectly? the general multipart form of the file name is part-{shard_num}-{file_num} .
an example full s3 path for a single file would look like folder/table/2024-06-03--22/part-0-182 .

![image](https://github.com/ClickHouse/ClickHouse/assets/12278781/6524201e-e9de-4062-a4d9-ac58ef771005)


https://github.com/ClickHouse/ClickHouse/tree/master/src/Storages/S3Queue

Maybe I'm blind but I dont really see the logic of how it discovers new files here. Is that in keeper? A point in the right direction would be nice :)

**Comment by deltarod** [comment]
#64349 fixes. Closing.
